doc: Update PubSub - Azure Event Hub for clarity (#2772)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-azure-eventhubs.md

| partitionCount | N | Number of partitions for the new event hub. Only used when entity management is enabled. Default: `"1"` | `"2"`
| messageRetentionInDays | N | Number of days to retain messages for in the newly created event hub. Used only when entity management is enabled. Default: `"1"` | `"90"`

## Component format

> To setup Azure Event Hubs pubsub create a component of type `pubsub.azure.eventhubs`. 

Reviewer: ```suggestion
To setup Azure Event Hubs pubsub, create a component of type `pubsub.azure.eventhubs`. 
```

> See [this guide]({{< ref "howto-publish-subscribe.md#step-1-setup-the-pubsub-component" >}}) on how to create and apply a pubsub configuration.

{{% alert title="Topic Name" color="primary" %}}

> Dapr requires the notion of topics. However, in a PubSub systems this does not exists (as it uses partitions instead). To work around this, Dapr utilizes the **Event Hub Name as the Topic Name** which will be found in the `topic` field in the event.

Reviewer 2: Dapr requires the notion of topics, a concept doesn't exist in Azure Event Hubs (partitions are used instead). To work around this, Dapr utilizes the **Event Hub Name as the Topic Name** which will be found in the `topic` field in the event.

{{% /alert %}}

The Azure Event Hubs pubsub component supports two authentication mechanisms: 
* Connection String
* [Azure AAD Authentication]({{< ref authenticating-azure.md >}})

{{% alert title="Entity Management" color="primary" %}}

> When using AAD Authentication, Dapr also supports **Entity Management**. This means that when configured correctly, Dapr will manipulate the Event Hub Namespace, creation of Event Hubs and Consumer Groups.

Reviewer 2: When using AAD Authentication, Dapr also supports **Entity Management**. When configured correctly, entity management allows Dapr to manipulate the Event Hub Namespace, creation of Event Hubs and Consumer Groups.

{{% /alert %}}

## Getting Started

To get started, follow these steps to use the Azure Event Hub component

{{% alert title="Warning" color="warning" %}}

> It is important to follow the creation of an Azure Event Hub account correctly as Dapr requires certain aspects to be configured.

Reviewer: ```suggestion
Make sure you follow the creation of an Azure Event Hub account correctly, as Dapr requires certain aspects to be configured.
```

{{% /alert %}}

### 1. Create an Azure Storage Account

> Dapr utilizes the Event Processor Host so an Azure Storage Account is required:

Reviewer: ```suggestion
Since Dapr utilizes the `Event Processor Host`, an Azure Storage Account is required:
```

> 1. Create an [Azure Storage Account](https://docs.microsoft.com/azure/storage/common/storage-account-create?tabs=azure-portal) resource (required since Dapr uses the `Event Processor Host`)

Reviewer: ```suggestion
1. Create an [Azure Storage Account](https://docs.microsoft.com/azure/storage/common/storage-account-create?tabs=azure-portal) resource.
```

> 2. [Copy the Storage Account Keys](https://docs.microsoft.com/azure/storage/common/storage-account-keys-manage)

Reviewer: ```suggestion
1. [Copy the Storage Account Keys](https://docs.microsoft.com/azure/storage/common/storage-account-keys-manage).
```

### 2. Create and Configure Azure Event Hubs

> 1. Create an [Azure Event Hub Namespace](https://docs.microsoft.com/azure/event-hubs/event-hubs-create) resource

Reviewer: ```suggestion
1. Create an [Azure Event Hub Namespace](https://docs.microsoft.com/azure/event-hubs/event-hubs-create) resource.
```

> 2. Create an `Event Hub` in the created namespace from step 1

Reviewer: ```suggestion
1. In the namespace resource you just created, create an `Event Hub`.
```

> 3. Create a Consumer Group for each app that wants to subscribe

Reviewer: ```suggestion
1. Create a Consumer Group for each app that wants to subscribe. The name of this Consumer Group equals the app id passed by either the:
```

>    1. The name of this consumer group equals the app id passed by the Dapr Run CLI flag `--app-id` or Kubernetes flag `dapr.io/app-id` (e.g. `dapr run --app-id example` or `dapr.io/app-id: "example"` requires a consumer group named `example`)

Reviewer: ```suggestion
   -  Dapr Run CLI flag `--app-id` (`dapr run --app-id <example>`), or
   - Kubernetes flag `dapr.io/app-id`: (`dapr.io/app-id: "<example>"`)
```

> 4. Decide which authentication mechanism to use (see example below)

Reviewer: ```suggestion
1. Decide which authentication mechanism to use:
```

   1. [Azure Authentication]({{< ref authenticating-azure.md >}})

>    2. [Connection String]((https://docs.microsoft.com/azure/event-hubs/authorize-access-shared-access-signature)) (copy the Connection String for the Event Hub from the Namespace or create a Shared Access Policy on the event hub itself (this will then include `EntityPath={EventHub}`).

Reviewer: ```suggestion
   1. [Connection String]((https://docs.microsoft.com/azure/event-hubs/authorize-access-shared-access-signature)); either:
      - Copy the Connection String for the Event Hub from the Namespace, or 
      - Create a Shared Access Policy on the Event Hub itself (this will then include `EntityPath={EventHub}`).
```

{{% alert title="Warning" color="warning" %}}

> When setting up an Azure Event Hub with the basic sku you will only have 1 consumer group available (named `$Default`). This means your app will be limited to using `--app-id '$Default'`!

Reviewer: ```suggestion
When setting up an Azure Event Hub with the basic sku, you will only have one consumer group named `$Default` available, limiting your app to using `--app-id '$Default'`.
```

{{% /alert %}}

## Azure IoT Hub Support

> Azure IoT Hub provides an [endpoint that is compatible with Event Hubs](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-read-builtin#read-from-the-built-in-endpoint), so the Azure Event Hubs pubsub component can also be used to subscribe to Azure IoT Hub events.

Reviewer: ```suggestion
Azure IoT Hub provides an [endpoint compatible with Event Hubs](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-read-builtin#read-from-the-built-in-endpoint), enabling the Azure Event Hubs pubsub component to be used to subscribe to Azure IoT Hub events.
```

### IoT Hub System Properties

> Beside only sending data, IoT Hub `device-to-cloud events` also contain additional [IoT Hub System Properties](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-construct#system-properties-of-d2c-iot-hub-messages). Dapr supports these and will return the following as part of the response metadata:

Reviewer: ```suggestion
Besides sending data, IoT Hub `device-to-cloud events` also contain additional [IoT Hub System Properties](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-messages-construct#system-properties-of-d2c-iot-hub-messages). Dapr supports these and will return the following as part of the response metadata:
```
